src/kbmod/work_unit.py

```diff
@@ -0,0 +1,170 @@
+                msk = hdu_to_raw_image(hdul[ext_num + 2])
+                if msk is None:
+                    msk = RawImage(np.zeros((sci.get_height(), sci.get_width())))
+
+                # Check if the PSF layer exists.
+                if hdul[ext_num + 3].header["NAXIS"] == 2:
+                    p = PSF(hdul[ext_num + 3].data)
+                else:
+                    p = PSF(1e-8)
+
+                imgs.append(LayeredImage(sci, var, msk, p))
+
+            im_stack = ImageStack(imgs)
+
+            result = WorkUnit(im_stack=im_stack, config=config)
+        return result
+
+    def write_to_file(self, filename, overwrite=False):
+        """Write the WorkUnit to a single file.
+
+        Parameters
+        ----------
+        filename : `str`
+            The file to which to write the data.
+        overwrite : bool
+            Indicates whether to overwrite an existing file.
+        """
+        if Path(filename).is_file() and not overwrite:
+            print(f"Warning: WorkUnit file {filename} already exists.")
+            return
+
+        # Set up the initial HDU list, including the primary header
+        # the metadata (empty), and the configuration.
+        hdul = fits.HDUList()
+        pri = fits.PrimaryHDU()
+        pri.header["NUMIMG"] = self.im_stack.img_count()
+        hdul.append(pri)
+        hdul.append(fits.BinTableHDU())
+        hdul.append(fits.BinTableHDU(self.config.to_table(make_fits_safe=True)))
+
+        for i in range(self.im_stack.img_count()):
+            layered = self.im_stack.get_single_image(i)
+            hdul.append(raw_image_to_hdu(layered.get_science()))
+            hdul.append(raw_image_to_hdu(layered.get_variance()))
+            hdul.append(raw_image_to_hdu(layered.get_mask()))
+
+            p = layered.get_psf()
+            psf_array = np.array(p.get_kernel()).reshape((p.get_dim(), p.get_dim()))
+            hdul.append(fits.hdu.image.ImageHDU(psf_array))
+
+        hdul.writeto(filename)
+
+
+def raw_image_to_hdu(img):
+    """Helper function that creates a HDU out of RawImage.
+
+    Parameters
+    ----------
+    img : `RawImage`
+        The RawImage to convert.
+
+    Returns
+    -------
+    hdu : `astropy.io.fits.hdu.image.ImageHDU`
+        The image extension.
+    """
+    # Expensive copy. To be removed with RawImage refactor.
+    np_pixels = np.array(img.get_all_pixels()).astype("float32", casting="same_kind")
+    np_array = np_pixels.reshape((img.get_height(), img.get_width()))
+    hdu = fits.hdu.image.ImageHDU(np_array)
+    hdu.header["MJD"] = img.get_obstime()
+    return hdu
+
+
+def hdu_to_raw_image(hdu):
+    """Helper function that creates a RawImage from a HDU.
+
+    Parameters
+    ----------
+    hdu : `astropy.io.fits.hdu.image.ImageHDU`
+        The image extension.
+
+    Returns
+    -------
+    img : `RawImage` or None
+        The RawImage if there is valid data and None otherwise.
+    """
+    img = None
+    if hdu.header["NAXIS"] == 2:
```

Comment:
I think it's better to be checking `if isinstance(hdu, (ImageHDU, CompImageHDU)` because those are the two image HDU types.